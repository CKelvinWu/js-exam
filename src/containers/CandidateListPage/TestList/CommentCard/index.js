import React from 'react';
import { Col, Card, Rate, Input } from 'antd';

const { TextArea } = Input;

class CommentCard extends React.Component {
  render() {
    return (
      <>
        <Col span={11}>
          <h3>Interviewer:</h3>
          <Card>
            <h4>Technical Skills：</h4>
            <div style={{ textAlign: 'right' }}>
              <h4 style={{ display: 'inline', marginRight: '12px' }}>
                Logical
              </h4>
              <Rate
                type="flex"
                align="middle"
                justify="space-around"
                defaultValue={3}
              />
            </div>
            <div style={{ textAlign: 'right' }}>
              <h4 style={{ display: 'inline', marginRight: '12px' }}>
                JavaScript Familiarity
              </h4>
              <Rate
                type="flex"
                align="middle"
                justify="space-around"
                defaultValue={3}
              />
            </div>
            <TextArea placeholder="Write your comments..." />
            <h4 style={{ marginTop: '12px' }}>Personality：</h4>
            <div style={{ textAlign: 'right' }}>
              <h4 style={{ display: 'inline', marginRight: '12px' }}>
                Enjoy to work with
              </h4>
              <Rate
                type="flex"
                align="middle"
                justify="space-around"
                defaultValue={3}
              />
            </div>
            <TextArea placeholder="Write your comments..." />
          </Card>
        </Col>
      </>
    );
  }
}

export default CommentCard;
