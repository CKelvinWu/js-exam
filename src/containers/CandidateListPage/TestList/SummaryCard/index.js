import React from 'react';
import { Row, Col, Card, Rate } from 'antd';
import PropTypes from 'prop-types';

class SummaryCard extends React.Component {
  render() {
    const { testListData } = this.props;
    return (
      <>
        <Col span={16}>
          <Row type="flex" align="middle" justify="space-between">
            <h3>Interviewer：{testListData[1]}</h3>
          </Row>
          <Card>
            <h4>Technical Skills：</h4>
            <div style={{ textAlign: 'right' }}>
              <h4 style={{ display: 'inline', marginRight: '12px' }}>
                Logical
              </h4>
              <Rate defaultValue={3} disabled />
            </div>
            <div style={{ textAlign: 'right' }}>
              <h4 style={{ display: 'inline', marginRight: '12px' }}>
                JavaScript Familiarity
              </h4>
              <Rate defaultValue={3} disabled />
            </div>
            <Card>
              <p>Card content</p>
            </Card>
            <h4 style={{ marginTop: '12px' }}>Personality：</h4>
            <div style={{ textAlign: 'right' }}>
              <h4 style={{ display: 'inline', marginRight: '12px' }}>
                Enjoy to work with
              </h4>
              <Rate defaultValue={3} disabled />
            </div>
            <Card>
              <p>Card content</p>
            </Card>
          </Card>
        </Col>
      </>
    );
  }
}

SummaryCard.propTypes = {
  testListData: PropTypes.array,
};

export default SummaryCard;
