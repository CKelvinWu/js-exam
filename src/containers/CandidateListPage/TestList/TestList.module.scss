.avatar {
  background-color: white;
  color: rgb(102, 102, 102);
}
.floatTop {
  background: none;
  font-size: 16px;
  border: none;
  cursor: pointer;
}
.floatTop:hover {
  transition: all 0.1s ease-in-out;
  transform: scale(1.3);
}
.delAnime {
  max-height: 100px;
  animation-name: delAnime;
  animation-duration: 0.5s;
  animation-timing-function: ease-in;
  animation-fill-mode: forwards;
  overflow: hidden;
}
@keyframes delAnime {
  from {
    max-height: 100px;
  }
  to {
    max-height: 0;
    padding: 0;
    border: rgba(255, 255, 255, 0);
  }
}
