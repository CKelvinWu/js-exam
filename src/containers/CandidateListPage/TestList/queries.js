export const getTest = `query GetTest($id: ID! $limit: Int) {
    getTest(id: $id) {
      id
      subjectId
      timeBegin
      timeEnd
      results(limit: $limit) {
        items {
          logic
          language
          workwith
          techreview
          perstyreview
          author
          role
        }
        nextToken
      }
      records(limit: $limit) {
        items {
          id
          subjectId
          syncCode
          timeBegin
          timeEnd
          comment {
            items {
              author
              time
              quality
              hint
              completeness
              content
            }
            nextToken
          }
          ques {
            type 
            content
            name
            test
          }
        }
        nextToken
      }
    }
  }`;
