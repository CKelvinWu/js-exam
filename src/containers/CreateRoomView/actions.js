/*
 *
 * Actions change things in your application
 * Since this boilerplate uses a uni-directional data flow, specifically redux,
 * we have these actions which are the only way your application interacts with
 * your application state. This guarantees that your state is up to date and nobody
 * messes it up weirdly somewhere.
 *
 * To add a new Action:
 * 1) Import your constant
 * 2) Add a function like this:
 *    export function yourAction(var) {
 *        return { type: YOUR_ACTION_CONSTANT, var: var }
 *    }
 */
import { API, graphqlOperation } from 'aws-amplify';
import * as mutations from 'graphql/mutations';
import { action } from 'utils/actionsHelper';
import { message } from 'antd';
import { CREATE_ROOM } from './constants';

export const createRoomActions = {
  request: data => action(CREATE_ROOM.REQUEST, { data }),
  success: data => action(CREATE_ROOM.SUCCESS, { data }),
  failure: error => action(CREATE_ROOM.FAILURE, { error }),
};

export function createRoom(data) {
  return async dispatch => {
    dispatch(createRoomActions.request(data));
    try {
      const { data: testData } = await API.graphql(graphqlOperation(mutations.createTest, data));
      const roomNum = Math.floor(Math.random() * 98) + 1;
      const roomChar = String.fromCharCode(Math.floor(Math.random() * 26) + 65);
      const { data: resData } = await API.graphql(
        graphqlOperation(mutations.createRoom, {
          input: {
            roomTestId: testData.createTest.id,
            description: roomChar + roomNum,
            createTime: new Date(),
            ...data.input,
          },
        }),
      );
      dispatch(createRoomActions.success(resData.createRoom));
    } catch (error) {
      dispatch(createRoomActions.failure(error));
      message.error('Create room failed');
    }
  };
}
