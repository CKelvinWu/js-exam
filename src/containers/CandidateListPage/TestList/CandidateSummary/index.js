import React from 'react';
import PropTypes from 'prop-types';
import { API, graphqlOperation } from 'aws-amplify';
import { Row, Col, Rate, Divider } from 'antd';

import { getTest } from '../../../../graphql/queries';
import CommentCard from '../CommentCard';

class CandidateSummary extends React.Component {
  // async getTestItemById(item) {
  //   const {
  //     data: { getTest: test },
  //   } = await API.graphql(graphqlOperation(getTest, { id: item.id }));
  //   return test;
  // }

  render() {
    const { testListData } = this.props;
    // console.log(testListData);
    // const data = this.getTestItemById(item);
    return (
      <>
        {/* {data.then(obj => {
          console.log(typeof obj);
        })} */}

        <Row type="flex" align="middle">
          <Col span={8} offset={3}>
            <h4>1. mergeSort</h4>
          </Col>
          <Col span={4}>
            <h4 style={{ display: 'inline' }}> Code quality </h4>
          </Col>
          <Col span={8}>
            <Rate disabled defaultValue={2} />
          </Col>
          <Col span={4} offset={11}>
            <h4 style={{ display: 'inline' }}> Compeleteness </h4>
          </Col>
          <Col span={8}>
            <Rate disabled defaultValue={2} />
          </Col>
          <Col span={4} offset={11}>
            <h4 style={{ display: 'inline' }}> How much hints </h4>
          </Col>
          <Col span={8}>
            <Rate disabled defaultValue={2} />
          </Col>
        </Row>
        <Divider dashed />
        {/* <Row type="flex" align="middle">
          <Col span={11} offset={2}>
            <h4>2. forEach, map and reduce</h4>
          </Col>
          <Col span={4}>
            <h4 style={{ display: 'inline' }}> Code quality </h4>
          </Col>
          <Col span={4}>
            <Rate disabled defaultValue={2} />
          </Col>
          <Col span={4} offset={13}>
            <h4 style={{ display: 'inline' }}> Compeleteness </h4>
          </Col>
          <Col span={4}>
            <Rate disabled defaultValue={2} />
          </Col>
          <Col span={4} offset={13}>
            <h4 style={{ display: 'inline' }}> How much hints </h4>
          </Col>
          <Col span={4}>
            <Rate disabled defaultValue={2} />
          </Col>
        </Row>
        <Divider dashed />
        <Row type="flex" align="middle">
          <Col span={11} offset={2}>
            <h4>3. pipe</h4>
          </Col>
          <Col span={4}>
            <h4 style={{ display: 'inline' }}> Code quality </h4>
          </Col>
          <Col span={4}>
            <Rate disabled defaultValue={2} />
          </Col>
          <Col span={4} offset={13}>
            <h4 style={{ display: 'inline' }}> Compeleteness </h4>
          </Col>
          <Col span={4}>
            <Rate disabled defaultValue={2} />
          </Col>
          <Col span={4} offset={13}>
            <h4 style={{ display: 'inline' }}> How much hints </h4>
          </Col>
          <Col span={4}>
            <Rate disabled defaultValue={2} />
          </Col>
        </Row>
        <Divider dashed /> */}
        <h2>Comments</h2>
        <CommentCard testListData={testListData} />
      </>
    );
  }
}
CandidateSummary.propTypes = {
  testListData: PropTypes.array,
};
export default CandidateSummary;
