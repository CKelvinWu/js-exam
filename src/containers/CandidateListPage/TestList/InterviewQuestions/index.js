import React from 'react';
import { API, graphqlOperation } from 'aws-amplify';
import { Col, Rate } from 'antd';

import { getTest } from '../../../../graphql/queries';

class InterviewQuestions extends React.Component {
  // async getTestItemById(item) {
  //   const {
  //     data: { getTest: test },
  //   } = await API.graphql(graphqlOperation(getTest, { id: item.id }));
  //   return test;
  // }

  render() {
    return (
      <>
        {/* {data.then(obj => {
          console.log(typeof obj);
        })} */}
        <Col span={8} offset={3}>
          <h4>1. mergeSort</h4>
        </Col>
        <Col span={4}>
          <h4 style={{ display: 'inline' }}> Code quality </h4>
        </Col>
        <Col span={8}>
          <Rate disabled defaultValue={2} />
        </Col>
        <Col span={4} offset={11}>
          <h4 style={{ display: 'inline' }}> Compeleteness </h4>
        </Col>
        <Col span={8}>
          <Rate disabled defaultValue={2} />
        </Col>
        <Col span={4} offset={11}>
          <h4 style={{ display: 'inline' }}> How much hints </h4>
        </Col>
        <Col span={8}>
          <Rate disabled defaultValue={2} />
        </Col>
        {/* <Row type="flex" align="middle">
          <Col span={11} offset={2}>
            <h4>2. forEach, map and reduce</h4>
          </Col>
          <Col span={4}>
            <h4 style={{ display: 'inline' }}> Code quality </h4>
          </Col>
          <Col span={4}>
            <Rate disabled defaultValue={2} />
          </Col>
          <Col span={4} offset={13}>
            <h4 style={{ display: 'inline' }}> Compeleteness </h4>
          </Col>
          <Col span={4}>
            <Rate disabled defaultValue={2} />
          </Col>
          <Col span={4} offset={13}>
            <h4 style={{ display: 'inline' }}> How much hints </h4>
          </Col>
          <Col span={4}>
            <Rate disabled defaultValue={2} />
          </Col>
        </Row>
        <Divider dashed />
        <Row type="flex" align="middle">
          <Col span={11} offset={2}>
            <h4>3. pipe</h4>
          </Col>
          <Col span={4}>
            <h4 style={{ display: 'inline' }}> Code quality </h4>
          </Col>
          <Col span={4}>
            <Rate disabled defaultValue={2} />
          </Col>
          <Col span={4} offset={13}>
            <h4 style={{ display: 'inline' }}> Compeleteness </h4>
          </Col>
          <Col span={4}>
            <Rate disabled defaultValue={2} />
          </Col>
          <Col span={4} offset={13}>
            <h4 style={{ display: 'inline' }}> How much hints </h4>
          </Col>
          <Col span={4}>
            <Rate disabled defaultValue={2} />
          </Col>
        </Row>
        <Divider dashed /> */}
      </>
    );
  }
}

export default InterviewQuestions;
