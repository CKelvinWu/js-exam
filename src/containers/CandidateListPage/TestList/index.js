import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { formatTime } from 'utils/format';

import { List, Avatar, Icon, Button, Modal, Tooltip } from 'antd';
import { deleteTestAction } from '../../../redux/test/actions';

import style from './TestList.module.scss';

import AddSummaryModal from 'components/Summary/AddSummaryModal';
import InterviewSummaryModal from 'components/Summary/InterviewSummaryModal';

class TestList extends React.Component {
  state = {
    delConfirmModalVisible: false,
    delTest: null,
    delAnime: false,
    testResultModalVisible: false,
    testResultModalTarget: '',
    addSummaryModalVisible: false,
    addSummaryModalTarget: '',
    testId: '',
  };

  shouldComponentUpdate(nextProps, nextState) {
    return !(this.state.delAnime && !nextState.delAnime);
  }

  handleDeleteButton = test => () => {
    this.setState({ delConfirmModalVisible: true, delTest: test });
  };

  hideDelConfirmModal = () => {
    this.setState({
      delConfirmModalVisible: false,
    });
  };

  handleOnOkDelConfirmModal = async () => {
    const { delTest } = this.state;
    const { deleteTestAction } = this.props;
    this.hideDelConfirmModal();
    // show the delete animation first and then do the delete action
    this.setState({ delAnime: true });
    setTimeout(async () => {
      await deleteTestAction(delTest);
      this.setState({ delAnime: false });
    }, 500);
  };

  showTestResultModal = e => {
    this.setState({
      testResultModalVisible: true,
      testResultModalTarget: e.target.getAttribute('candidate'),
      testId: e.target.getAttribute('testid'),
    });
  };

  testResultModalCancel = () => {
    this.setState({
      testResultModalVisible: false,
    });
  };

  showAddSummaryModal = e => {
    this.setState({
      addSummaryModalVisible: true,
      addSummaryModalTarget: e.target.getAttribute('candidate'),
      testId: e.target.getAttribute('testid'),
    });
  };

  addSummaryModalCancel = () => {
    this.setState({
      addSummaryModalVisible: false,
    });
  };

  toInterviewResult = data => {
    const interviewers = data.users.items.map(v => v.user);
    const questions = data.records.items.map(v => ({
      id: v.id,
      name: v.ques.name,
    }));
    const comments = [];
    data.records.items.forEach(r => {
      r.comment.items.forEach(c => {
        comments.push({
          questionID: r.id,
          ...c,
        });
      });
    });
    const summaries = data.results.items;
    return { interviewers, questions, comments, summaries };
  };

  render() {
    const { testListData } = this.props;
    const {
      delTest,
      delConfirmModalVisible,
      delAnime,
      testResultModalVisible,
      testResultModalTarget,
      addSummaryModalVisible,
      addSummaryModalTarget,
      testId,
    } = this.state;
    const jeUser = localStorage.jeUser && JSON.parse(localStorage.jeUser);

    return (
      <>
        <List
          itemLayout="horizontal"
          dataSource={testListData}
          renderItem={item => (
            <List.Item
              className={
                delAnime && delTest && delTest.id === item.id
                  ? style.delAnime
                  : ''
              }
              actions={[
                <Button
                  size="small"
                  onClick={this.showTestResultModal}
                  candidate={item.subjectId}
                  testid={item.id}
                >
                  Open Summary
                </Button>,
                <Link
                  to={{
                    pathname: `/admin/playback/${item.id}`,
                  }}
                >
                  Playback
                </Link>,
              ]}
            >
              <List.Item.Meta
                avatar={<Avatar icon="code" className={style.avatar} />}
                title={item.subjectId}
                description={formatTime(item.timeBegin)}
              />
              {item.host && item.host.name === localStorage.username && (
                <button
                  type="button"
                  className={style.floatTop}
                  onClick={this.handleDeleteButton(item)}
                >
                  <Icon type="delete" theme="twoTone" twoToneColor="#f00" />
                </button>
              )}
              {item.users.items &&
                item.users.items.map(v => v.user.id).includes(jeUser.id) &&
                !item.results.items
                  .map(v => v.author)
                  .includes(jeUser.name) && (
                  <Tooltip
                    placement="top"
                    title="write summary"
                    onClick={this.handleSummaryEdit}
                  >
                    <Button
                      type="link"
                      icon="edit"
                      style={{
                        fontSize: '18px',
                        cursor: 'pointer',
                        margin: '0 -20px 0 20px',
                      }}
                      candidate={item.subjectId}
                      testid={item.id}
                      onClick={this.showAddSummaryModal}
                    />
                  </Tooltip>
                )}
            </List.Item>
          )}
        />
        <InterviewSummaryModal
          testID={testId}
          title={`Candidateï¼š${testResultModalTarget}`}
          visible={testResultModalVisible}
          onCancel={this.testResultModalCancel}
          footer={null}
          width={1000}
        ></InterviewSummaryModal>
        <AddSummaryModal
          testID={testId}
          title={addSummaryModalTarget}
          visible={addSummaryModalVisible}
          onCancel={this.addSummaryModalCancel}
          footer={null}
          width={800}
        ></AddSummaryModal>
        <Modal
          title=""
          visible={delConfirmModalVisible}
          okType="danger"
          okText="Delete"
          onOk={this.handleOnOkDelConfirmModal}
          onCancel={this.hideDelConfirmModal}
        >
          Are you sure you want to delete test{' '}
          <b>{delTest ? delTest.subjectId : ''}</b> ?
        </Modal>
      </>
    );
  }
}
TestList.propTypes = {
  testListData: PropTypes.array,
};

const mapDispatchToProps = dispatch => ({
  deleteTestAction: delTest => dispatch(deleteTestAction(delTest)),
});
export default connect(null, mapDispatchToProps)(TestList);
