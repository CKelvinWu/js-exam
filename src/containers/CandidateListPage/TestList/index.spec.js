import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import BrowserRouter from 'react-router-dom/BrowserRouter';
import { List } from 'antd';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import TestList from './index';
import * as deleteActionUtils from '../../../redux/test/actions';

describe('TestList index', () => {
  const tests = [
    {
      id: '9faebbd5-936e-4d3f-8345-ae63c100c9d7',
      subjectId: 'Zone',
      description: null,
      timeBegin: '2020-01-17T04:02:09.202Z',
      tags: ['Hank'],
    },
    {
      id: 'ea52ba9d-5d72-4ac2-8e22-ac4570264706',
      subjectId: 'testerror',
      description: null,
      timeBegin: '2020-01-17T02:35:53.776Z',
      tags: ['Kenny'],
    },
  ];

  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const initialState = {};
  const testStore = mockStore(initialState);
  let wrapper;
  let list;
  let renderedItem;
  let deleteButton;

  function withProvider(component) {
    return mount(
      <Provider store={testStore}>
        <BrowserRouter>{component}</BrowserRouter>
      </Provider>,
    );
  }

  beforeAll(() => {
    localStorage.setItem('username', 'Kenny');
    wrapper = withProvider(<TestList data={tests} />);
    list = wrapper.find('TestList').instance();
    renderedItem = wrapper.find(List.Item.Meta);
    deleteButton = wrapper.find('.floatTop');
  });

  beforeEach(() => {
    testStore.clearActions();
    jest.clearAllMocks();
  });

  it('should render TestList', () => {
    expect(wrapper.render()).toMatchSnapshot();
    expect(renderedItem.length).toEqual(2);
    expect(deleteButton.length).toEqual(1);
  });

  it('should handleDeleteButton and hideDelConfirmModal properly', () => {
    deleteButton.simulate('click');
    expect(list.state.delConfirmModalVisible).toBeTruthy();
    expect(list.state.delTest).toEqual(tests[1]);
    list.hideDelConfirmModal();
    expect(list.state.delConfirmModalVisible).toBeFalsy();
  });

  it('should handleOnOkDelConfirmModal', async () => {
    jest.useFakeTimers();
    list.state.delTest = 'atest';
    const theMock = jest.fn(delTest => jest.fn());
    deleteActionUtils.deleteTestAction = theMock;

    const hideDelConfirmSpy = jest.spyOn(list, 'hideDelConfirmModal');
    await list.handleOnOkDelConfirmModal().catch();
    expect(hideDelConfirmSpy).toHaveBeenCalled();
    expect(list.state.delAnime).toBeTruthy();

    jest.advanceTimersByTime(550);
    await wrapper.update();
    expect(theMock).toHaveBeenCalledWith('atest');
    expect(list.state.delAnime).toBeFalsy();
  });
});
