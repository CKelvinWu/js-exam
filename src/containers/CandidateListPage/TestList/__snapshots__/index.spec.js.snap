// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestList index should render TestList 1`] = `
<div
  class="ant-list ant-list-split"
>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <ul
        class="ant-list-items"
      >
        <li
          class="ant-list-item"
        >
          <div
            class="ant-list-item-meta"
          >
            <div
              class="ant-list-item-meta-avatar"
            >
              <span
                class="ant-avatar avatar ant-avatar-circle ant-avatar-icon"
              >
                <i
                  aria-label="icon: code"
                  class="anticon anticon-code"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="code"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M516 673c0 4.4 3.4 8 7.5 8h185c4.1 0 7.5-3.6 7.5-8v-48c0-4.4-3.4-8-7.5-8h-185c-4.1 0-7.5 3.6-7.5 8v48zm-194.9 6.1l192-161c3.8-3.2 3.8-9.1 0-12.3l-192-160.9A7.95 7.95 0 0 0 308 351v62.7c0 2.4 1 4.6 2.9 6.1L420.7 512l-109.8 92.2a8.1 8.1 0 0 0-2.9 6.1V673c0 6.8 7.9 10.5 13.1 6.1zM880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
                    />
                  </svg>
                </i>
              </span>
            </div>
            <div
              class="ant-list-item-meta-content"
            >
              <h4
                class="ant-list-item-meta-title"
              >
                Zone
              </h4>
              <div
                class="ant-list-item-meta-description"
              >
                2020/01/17 12:02:09
              </div>
            </div>
          </div>
          <ul
            class="ant-list-item-action"
          >
            <li>
              <button
                candidate="Zone"
                class="ant-btn ant-btn-sm"
                interviewer="Hank"
                type="button"
              >
                <span>
                  Open Summary
                </span>
              </button>
              <em
                class="ant-list-item-action-split"
              />
            </li>
            <li>
              <a
                href="/admin/playback/9faebbd5-936e-4d3f-8345-ae63c100c9d7"
              >
                Playback
              </a>
            </li>
          </ul>
        </li>
        <li
          class="ant-list-item"
        >
          <div
            class="ant-list-item-meta"
          >
            <div
              class="ant-list-item-meta-avatar"
            >
              <span
                class="ant-avatar avatar ant-avatar-circle ant-avatar-icon"
              >
                <i
                  aria-label="icon: code"
                  class="anticon anticon-code"
                >
                  <svg
                    aria-hidden="true"
                    class=""
                    data-icon="code"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M516 673c0 4.4 3.4 8 7.5 8h185c4.1 0 7.5-3.6 7.5-8v-48c0-4.4-3.4-8-7.5-8h-185c-4.1 0-7.5 3.6-7.5 8v48zm-194.9 6.1l192-161c3.8-3.2 3.8-9.1 0-12.3l-192-160.9A7.95 7.95 0 0 0 308 351v62.7c0 2.4 1 4.6 2.9 6.1L420.7 512l-109.8 92.2a8.1 8.1 0 0 0-2.9 6.1V673c0 6.8 7.9 10.5 13.1 6.1zM880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
                    />
                  </svg>
                </i>
              </span>
            </div>
            <div
              class="ant-list-item-meta-content"
            >
              <h4
                class="ant-list-item-meta-title"
              >
                testerror
              </h4>
              <div
                class="ant-list-item-meta-description"
              >
                2020/01/17 10:35:53
              </div>
            </div>
          </div>
          <button
            class="floatTop"
            type="button"
          >
            <i
              aria-label="icon: delete"
              class="anticon anticon-delete"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="delete"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M292.7 840h438.6l24.2-512h-487z"
                  fill="#ffeae6"
                />
                <path
                  d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-504-72h304v72H360v-72zm371.3 656H292.7l-24.2-512h487l-24.2 512z"
                  fill="#f00"
                />
              </svg>
            </i>
          </button>
          <ul
            class="ant-list-item-action"
          >
            <li>
              <button
                candidate="testerror"
                class="ant-btn ant-btn-sm"
                interviewer="Kenny"
                type="button"
              >
                <span>
                  Open Summary
                </span>
              </button>
              <em
                class="ant-list-item-action-split"
              />
            </li>
            <li>
              <a
                href="/admin/playback/ea52ba9d-5d72-4ac2-8e22-ac4570264706"
              >
                Playback
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
